src/main/java/net/minecraftforge/fml/common/eventhandler/Event.java

```diff
@@ -90,7 +91,10 @@ public void setCanceled(boolean cancel)
     {
         if (!isCancelable())
         {
-            throw new IllegalArgumentException("Attempted to cancel a non cancellable event");
+            throw new UnsupportedOperationException(
+                "Attempted to modify the unmodifiable canceled state of a non-cancelable event of type: "
```

Comment:
I think the original wording is much more clear. Just add the class name to the end.